### DISCUSSION TITLE
Group and Individual Parameters in Regression

### DISCUSSION THREAD

**Discussion by JoeSu112** [lnccbrown/HSSM#349] [Q&A] [answered]
Hi all, @frankmj, I am now trying to transition my data from HDDM to HSSM. I've asked about the interpretation of group- and subject-level parameters for `a ~ 1 + (1|subject) in the model (#345). However, I have new problems on the interpretation of group- and subject-level parameters in regression:

I put `v ~ 1 + (beta1|subject) + (beta2|subject) + (beta3|subject)` in the model, so that the drift rate is a combination of the intercept, beta1, beta2 and beta3. It's a bit different from above. The result table is as followed:

| Header | Mean|
|--------|--------|
| v_Intercept | 0.09406|
| v_1｜subject[1]| 0.09397|
| v_1｜subject[2]|-0.03487|
| v_beta1｜subject[1] |0.29647  |
| v_beta1｜subject[2] | 0.21175 |
| v_beta2｜subject[1] | 0.22109 |
| v_beta2｜subject[2] |0.08195 |
| v_beta3｜subject[1]| -0.18794| 
| v_beta3｜subject[2]| 0.55229 | 

There are v_Intercept, v_1｜subject[i], v_beta1｜subject[i], v_beta1｜subject[i], v_beta1｜subject[i] in the table. And I am wondering how I can get group and individual parameters of the intercept, beta1, beta2 and beta3. 
Thanks for any help!!!

**Comment by gpagnier** [comment, marked as answer]
Hi,
There's a tutorial in the works that will address these questions but in the meantime I'll do my best to answer. When creating these models its generally a good idea to plot the graph of the model using graph() to see exactly what group/individual parameters you're asking the model to fit. If it's truly hierarchical, you can see this with graph() where there should be a mean and std (sigma) for the individual parameters. 
In the equation you posted, it looks like you're only asking for a group effect of the intercept (represented by v_intercept). v_1|subject[1] represents the individual offset for subject[1] so the **individual intercept** for subject 1 in the example above would be **v_Intercept + v_1|subject[1]**. To get for the full group and individual effects for beta1 and beta2 you should define your regression equation to be:
v ~ 1 + beta1 + beta2 + (1|subject) + (beta1|subject) + (beta2|subject). 
If you set hierarchical to True, HSSM will automatically turn any unspecified parameter into a regression with the terms 1 + (1|participant_id) . Remember that if you use this, you need to have a column denoting participant ids called 'participant_id'.
